agent/consul/server.go

```diff
@@ -1400,6 +1402,10 @@ func (s *Server) setupExternalGRPC(config *Config, deps Deps, logger hclog.Logge
 }
 
 func (s *Server) setupInsecureResourceServiceClient(typeRegistry resource.Registry, logger hclog.Logger) error {
+	if s.raftStorageBackend == nil {
```

Comment:
Are you fine with the resource server being nil in the insecure client?

Reply:
Yes, the resource server is not used to set up that client. `setupInsecureResourceServiceClient` function creates a separate server with no auth and so it should be Ok for the one on the server struct to be nil